### DISCUSSION TITLE
Kernel launch needs more blocks

### DISCUSSION THREAD

**Discussion by PgLoLo** [jax-ml/jax#6814] [Q&A]
I have just got this strange exception: 
```RuntimeError: Unimplemented: Kernel launch needs more blocks (2690746531) than allowed by hardware (2147483647).``` 
I think it's somehow related to the `jax.scipy.optimize.minimize`. Have never seen this error before and as it should be somehow related to the low-level implementation of Jax, I won't bother to make some meaningful reproducer for now. 

Any idea what is it and how it could be work-arouned?

**Comment by jakevdp** [comment]
There is some discussion that may be relevant here: https://github.com/google/jax/issues/5733
